### PR TITLE
Add REST tests for histogram aggregation

### PR DESCRIPTION
**Pull request by csoulios**
Added more tests for histogram aggregation
Related to #26220

### CODE REVIEW

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.aggregation/10_histogram.yml
Comment (nik9000) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.aggregation/10_histogram.yml:129]: I tend to use `bulk` for these just to save a bit of typing. Maybe it faster, but it makes the tests more likely to fit onto one screen. Even refreshing is just one line in the bulk.

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.aggregation/10_histogram.yml
Comment (nik9000) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.aggregation/10_histogram.yml:158]: I tend not to add these because I think we want to remove them from master in the end. They are useful on backport though. Either way is cool.

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.aggregation/10_histogram.yml
Comment (nik9000) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.aggregation/10_histogram.yml:159]: I think it's easier to read when these are on individual lines. I tend to write them in yaml instead of json too just because we're in yaml files. But I think that is less important than the line feeds.

**Review by nik9000 — APPROVED** [review]
Wonderful! I left a few minor things. I think the new lines point I left is important for consistency with the more modern yaml tests, but the others are all your call if you want to apply them. Either way, LGTM. No need for another review pass.

**Comment by csoulios** [comment]
Nik thanks a lot for the feedback. I applied all changes you suggested in 7319babbcf13f68a5160088af995c31682b07e88

**Review by nik9000 — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**
